src/core/Akka.Cluster/Cluster.cs

```diff
@@ -333,28 +333,27 @@ public void JoinSeedNodes(IEnumerable<Address> seedNodes)
         /// </summary>
         /// <param name="seedNodes">TBD</param>
         /// <param name="token">TBD</param>
-        public Task JoinSeedNodesAsync(IEnumerable<Address> seedNodes, CancellationToken token = default(CancellationToken))
+        public Task JoinSeedNodesAsync(IEnumerable<Address> seedNodes, CancellationToken token = default)
         {
-            var completion = new TaskCompletionSource<NotUsed>(TaskCreationOptions.RunContinuationsAsynchronously);
```

Comment:
Something we don't have today that we should add - a bailout condition in the event that we're already a part of the cluster. Either we should just return an immediate `Task.Completed` or an exception. I'd probably prefer to do the former and keep chugging along.

Reply:
I think it would be more polite to return Task.Completed, IIRC that is more or less what existing behavior is.

Reply:
That sounds like a better idea, I'll try to revamp this PR to fit around that idea instead.